### ISSUE TITLE
checkpoint_daemon.lua:49: bad argument #2 to 'format'

### ISSUE THREAD

**Issue by kmansoft** [tarantool/tarantool#4030]
Copying from

https://groups.google.com/forum/#!topic/tarantool-ru/xKSj20wcoPo

Tarantool version:

1.10.2-171-gafe89a115

OS version:

Debian 9.8 64 bit

Bug description:

Error messages in Tarantool log

Steps to reproduce:

Getting these after upgrading from 1.9 to 1.10

2019-03-06 07:13:31.489 [25904] main/107/checkpoint_daemon checkpoint_daemon.lua:87 E> builtin/box/checkpoint_daemon.lua:49: bad argument #2 to 'format' (number expected, got cdata)

**Comment by locker** [comment]
Looks like the problem is caused by big checkpoint signatures:
```
-rw-r--r-- 1 push push 13M Mar  6 00:10 00000000007482838489.snap
-rw-r--r-- 1 push push 13M Mar  6 02:13 00000136076343052970.snap
-rw-r--r-- 1 push push 13M Mar  6 09:32 00000136076347248456.snap
```

**Comment by locker** [comment]
2.1 shouldn't be affected as there's no checkpoint_daemon.lua (it was rewritten in C), but need to check. We may either backport that patch or fix checkpoint_daemon.lua.

**Comment by kmansoft** [comment]
Strange things:

1 - The LSN increased by 128 billion between 0:10 and 2:13 - this is when I upgraded from 1.9 to 1.10

I can't imagine that my code (even with theoretical bugs) - running on a 2 CPU VPS - really created 128 billion real write operations in 2 hours.

2 - From 2:13 to 9:32, there isn't a similar "outrageous" increase in LSN.

So the increase by 128 billions only happened once, after upgrading 1.9 to 1.10.

**Comment by kmansoft** [comment]
Snapshotting is now broken:

```
2019-03-06 11:13:31.492 [25904] main/107/checkpoint_daemon checkpoint_daemon.lua:87 E> builtin/box/checkpoint_daemon.lua:49: bad argument #2 to 'format' (number expected, got cdata)
2019-03-06 12:13:31.493 [25904] main/107/checkpoint_daemon checkpoint_daemon.lua:87 E> builtin/box/checkpoint_daemon.lua:49: bad argument #2 to 'format' (number expected, got cdata)
2019-03-06 13:13:31.493 [25904] main/107/checkpoint_daemon checkpoint_daemon.lua:87 E> builtin/box/checkpoint_daemon.lua:49: bad argument #2 to 'format' (number expected, got cdata)
```

and there are no new .snap files - and .xlog files keep proliferating.

Any suggestions on what to do now? The database is broken now, as there is no snapshotting anymore, right?

**Comment by locker** [comment]
Fixed by b3656d459244c193053ae6ad9686d1cbd875a6f1
